### ISSUE TITLE
Difference in Skeleton3D.get_bone_global_pose_no_override behavior for 4.1

### ISSUE THREAD

**Issue by uguu-org** [godotengine/godot#79136]
### Godot version

4.1.stable

### System information

Windows 10

### Issue description

In Godot 4.0.3 and earlier, Skeleton3D.get_global_bone_pose_no_override appears to take into account of all transformations and overrides of the parent bone, while in Godot 4.1, parent overrides are ignored.

Godot 4.0.3 behavior is more convenient for my use case, but I can't tell if it just has been wrong all this time, hence this bug.

### Steps to reproduce

1. Load the attached project in Godot 4.0.3.
2. Run the project, and observe how the transformations for the 3 cones are stacked:

![godot40](https://github.com/godotengine/godot/assets/2000051/13872f94-9f00-4608-8679-2a0b77164938)

3. Load and convert the same project in Godot 4.1
4. Run the project again, observe that all three cones have the same transformation:

![godot41](https://github.com/godotengine/godot/assets/2000051/51e756e8-b8bd-44ac-8d9c-aaf7131793b2)

See comments in lines 13-20 of skeleton_test/main.gd for where the behavior differs.

### Minimal reproduction project

[skeleton_test.zip](https://github.com/godotengine/godot/files/11973311/skeleton_test.zip)

**Comment by wagnerfs** [comment]
Was just about to file the same bug report, all code that relied on get_bone_global_pose_no_override are now broken.

![bones_issue](https://github.com/godotengine/godot/assets/106078/698990c3-ed2d-4bfa-a9cd-a046116a0420)

**Comment by lyuma** [comment]
In Godot 4.0.x, `get_bone_global_pose_no_override` mistakenly returned the same value as `get_bone_global_pose`. This bug has been corrected in 4.1 but it means that scripts which depended on this incorrect behavior will appear to have broken.
See the PR which was introduced for 4.1 here:
* #77194

The good news is this: if your code worked properly in Godot 4.0, that means your code is designed to work with `get_bone_global_pose` instead, so you should replace all calls to `get_bone_global_pose_no_override` with `get_bone_global_pose` and it should then work the same between Godot 4.0 and 4.1

In short, this is not a bug, but it is something we should have indicated more clearly in the release notes.

**Comment by wagnerfs** [comment]
@lyuma not quite, the issue there is that the last bone, the one being requested, needs to be `bone_global_pose_no_override`, `get_bone_global_pose` will return the chain global_pose INCLUDING the requested bone + the previous frame's override.
The way it worked on 4.0 is that we could get a "rest" pose for the requested bone but transformed by the chain behind that had all changes to them.

I was curious if that behavior was intended, so in that case we probably need a method that can return the above, the chain having all changes applied and the last bone on rest transform, otherwise we'd have to calculate the entire chain by hand.


https://github.com/godotengine/godot/assets/106078/1e224fb6-9435-4e61-9002-6840587e213d


As you can see here, changing `get_bone_global_pose_no_override` to `get_bone_global_pose` will completely unlink the bone from the current animation being played.

**Comment by wagnerfs** [comment]
@akien-mga alright, I checked and the change does completely ruin IK, although all previous calls to `bone_global_pose_no_override` will now have to be replaced by:

`skeleton3d.get_bone_global_pose(parent_bone) * skeleton3d.get_bone_pose(bone)`, so just leaving this here in case other people might come across this

**Comment by lyuma** [comment]
The explanation is way simpler. Read what I wrote.

This is how the code looked in 4.0:
```
$ cgrep -r pose_global_no_override scene
scene/3d/skeleton_3d.cpp:       return bones[p_bone].pose_global_no_override;
scene/3d/skeleton_3d.cpp:                               b.pose_global_no_override = b.pose_global;
scene/3d/skeleton_3d.cpp:                               b.pose_global_no_override = b.pose_global;
scene/3d/skeleton_3d.cpp:                               b.pose_global_no_override = b.pose_global;
scene/3d/skeleton_3d.cpp:                               b.pose_global_no_override = b.pose_global;
scene/3d/skeleton_3d.h:         Transform3D pose_global_no_override;
```
That is to say, `get_bone_pose_global_no_override()` quite literally returns the same thing as what `get_bone_pose_global()` returns both in 4.0 and 4.1

Therefore, to upgrade your project from 4.0 to 4.1, replace your `get_bone_pose_global_no_override()` calls with `get_bone_pose_global()`. That's it.

To demonstrate, I took your MRP project and did this change, and it solves the issue:

![image](https://github.com/godotengine/godot/assets/39946030/b0fc4af3-335e-4336-95aa-4bbb4192ae9b)

Can you try this instead of the `skeleton3d.get_bone_global_pose(parent_bone) * skeleton3d.get_bone_pose(bone)` operation you are attempting?

**Comment by wagnerfs** [comment]
@lyuma it's because you're not taking into consideration animations, in all my examples I'm doing changes to bones while the skeleton is being animated, that's why you can't see the problem here, both `get_bone_pose_global()` and `get_bone_pose_global_no_override()` will certainly return the same if the skeleton has no other influences.

**Comment by lyuma** [comment]
Ok, I see what you mean. My apologies for being wrong about this.
I missed that this logic was done after assigning `pose_global_no_override` in the old code.
```cpp
if (b.global_pose_override_amount >= CMP_EPSILON) {
	b.pose_global = b.pose_global.interpolate_with(b.global_pose_override, b.global_pose_override_amount);
}
```

So you are correct. To emulate what the old `get_bone_global_pose_no_override` returned, it's probably something like this:
```gd
var parent_bone = skeleton3d.get_bone_parent(bone)
var parent_transform: Transform3D
if parent_bone != -1:
	parent_transform = skeleton3d.get_bone_global_pose(parent_bone)
if skeleton3d.is_bone_enabled(bone):
	return parent_transform * skeleton3d.get_bone_pose(bone)
else:
	return parent_transform * skeleton3d.get_bone_rest(bone);
```

**Comment by uguu-org** [comment]
Applying the `skeleton.get_bone_global_pose(skeleton.get_bone_parent(bone)) * skeleton.get_bone_pose(bone)` change above fixed it for me, thanks!

**Comment by elXill** [comment]
I could finaly be able to control multiple bones in the same hierarchy with this `skeleton.get_bone_global_pose(skeleton.get_bone_parent(bone)) * skeleton.get_bone_pose(bone)` while it works perfectly fine now finding this info was really hard and took me a few days..
I wish this could be done with a more straight forward way, even now I'm not sure what does multiplying Transform3D's do but I'm glad it works.
